static/js/home-try-it.js

```diff
@@ -0,0 +1,186 @@
+                    type_name: type_name,
+                    id: area.id
+                });
+            });
+
+            return newAreas;
+        }
+
+        var displayPostcode = function displayPostcode(data) {
+            var context = {
+                postcode: data.postcode,
+                areas: extractAreas(data.areas)
+            };
+
+            $(renderTemplate('try-result', context)).insertAfter($try);
+
+            var map = new L.Map("homepage-leaflet", {
+                scrollWheelZoom: false
+            });
+            map.attributionControl.setPrefix('');
+
+            var point = new L.LatLng(data.wgs84_lat, data.wgs84_lon);
+            map.setView(point, 14);
+
+            var layers = {
+                osm: new L.TileLayer('http://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png', {
+                    attribution: 'Map © <a href="https://www.openstreetmap.org/copyright">OpenStreetMap</a> contributors',
+                    maxZoom: 14,
+                    minZoom: 4
+                }),
+                marker: new L.Marker(point)
+            };
+            map.addLayer(layers.marker);
+            map.addLayer(layers.osm);
+
+            // Find a smallish area, and pre-select it.
+            var lsoa = _.findWhere(context.areas, {type: 'OLF'});
+            if (lsoa) {
+                addArea(map, layers, lsoa.id);
+                $('[data-areaid="' + lsoa.id + '"]').addClass('selected');
+            }
+
+            $('.homepage-try-result__area').on('click', function(e){
+                e.preventDefault();
+                var id = $(this).data('areaid');
+
+                if (id in layers) {
+                    removeArea(map, layers, id);
+                    $(this).removeClass('selected');
+                } else {
+                    $(this).siblings('.selected').each(function(){
+                        removeArea(map, layers, $(this).data('areaid'));
+                        $(this).removeClass('selected');
+                    });
+                    addArea(map, layers, id);
+                    $(this).addClass('selected');
+                }
+            });
+
+            $('.homepage-try-result__close').on('click', function(){
+                $('.homepage-try-result').slideUp(250);
+                $('.homepage-try form input[type="text"]').val('');
+            });
+        }
+
+        var removeArea = function removeArea(map, layers, area_id){
+            if (area_id in layers) {
+                map.removeLayer( layers[area_id] );
+                delete layers[area_id];
+            }
+        }
+
+        var addArea = function addArea(map, layers, area_id){
+            $.ajax({
+                dataType: 'json',
+                url: '/area/' + area_id + '.geojson',
+                data: {
+                    simplify_tolerance: '0.0001'
+                }
+
+            }).done(function(geojson){
+                layers[area_id] = new L.GeoJSON(geojson, {
```

Comment:
Even if you make it stupid high, Scotland still takes forever (it just has so many points) and looks ridiculous :) Though I guess on live site, the boundaries will be cached for weeks which might help. Especially if we can get some form of stale-while-revalidate going and request them every so often ourselves...